src/Microsoft.Health.Dicom.SqlServer/Features/Query/SqlQueryGenerator.cs

```diff
@@ -121,6 +132,85 @@ private void AppendFilterTable(string filterAlias)
             _stringBuilder.AppendLine($") {filterAlias}");
         }
 
+        private void AppendCustomTagTables()
+        {
+            Dictionary<CustomTagDataType, CustomTagLevel> queriedCustomTagDataTypesPerLevel = new Dictionary<CustomTagDataType, CustomTagLevel>();
+            foreach (CustomTagFilterDetails filterDetails in _queryExpression.QueriedCustomTagFilterDetails)
+            {
+                CustomTagDataType dataType;
+                CustomTagLimit.CustomTagVRAndDataTypeMapping.TryGetValue(filterDetails.Tag.GetDefaultVR().Code, out dataType);
+                if (!queriedCustomTagDataTypesPerLevel.TryAdd(dataType, filterDetails.Level) && queriedCustomTagDataTypesPerLevel[dataType] > filterDetails.Level)
+                {
+                    queriedCustomTagDataTypesPerLevel[dataType] = filterDetails.Level;
+                }
+            }
+
+            foreach (CustomTagFilterDetails filterDetails in _queryExpression.QueriedCustomTagFilterDetails)
+            {
+                CustomTagDataType dataType;
+                CustomTagLimit.CustomTagVRAndDataTypeMapping.TryGetValue(filterDetails.Tag.GetDefaultVR().Code, out dataType);
+                string customTagTableAlias = null;
+                _stringBuilder.Append("INNER JOIN ");
+                switch (dataType)
+                {
+                    case CustomTagDataType.StringData:
+                        customTagTableAlias = CustomTagStringTableAlias + filterDetails.Key;
+                        _stringBuilder.AppendLine($"{VLatest.CustomTagString.TableName} {customTagTableAlias}");
```

Comment:
How will this work for multiple tags on the same table? Will you not reuse the same alias which will fail the SQL syntax